src/transaction_builder.cpp

```diff
@@ -46,12 +51,17 @@ bool TransactionBuilder::AddSaplingSpend(
     return true;
 }
 
-void TransactionBuilder::AddSaplingOutput(
+bool TransactionBuilder::AddSaplingOutput(
```

Comment:
I've realised that there's a problem here: if you don't check the return value of the function, you can just carry on and call `Build()`. Because this PR does not update the RPC call sites to check the return value, the original problem remains. I'm now wondering if it's "safer" to throw an exception instead of returning false (since we don't yet have something like Rust's `Result<T, E>` for error-handling).

Reply:
I've made this change in #3646

Reply:
Resolved, now throws exception.